doc/modules/pipeline.rst

```diff
@@ -124,6 +119,48 @@ i.e. if the last estimator is a classifier, the :class:`Pipeline` can be used
 as a classifier. If the last estimator is a transformer, again, so is the
 pipeline.
 
+Caching transformers: avoid repeated computation
+-------------------------------------------------
+
+.. currentmodule:: sklearn.pipeline
+
+Fitting transformers may sometimes be computational expensive. With its
+``memory`` parameter set, class:`Pipeline` will cache each transformer
+after calling ``fit``.
+This feature is used to avoid computing the fit transformers within a pipeline
+if the parameters and input data are identical. A typical example is the case of
+a grid search in which the transformers can be fitted only once and reuse for
+each configuration.
+
+The parameter ``memory`` is needed in order to cache the transformers.
+``memory`` can be either a string containing the directory where to cache the
+transfomers or a `joblib.Memory<https://pythonhosted.org/joblib/memory.html>`_
+object::
+
+    >>> from tempfile import mkdtemp
+    >>> from shutil import rmtree
+    >>> from sklearn.decomposition import PCA
+    >>> from sklearn.svm import SVC
+    >>> from sklearn.pipeline import Pipeline
+    >>> param_grid = [('reduce_dim', PCA()), ('clf', SVC())]
+    >>> cachedir = mkdtemp()
+    >>> pipe = Pipeline(param_grid, memory=cachedir)
+    >>> pipe # doctest: +NORMALIZE_WHITESPACE, +ELLIPSIS
+    Pipeline(...,
+             steps=[('reduce_dim', PCA(copy=True,...)),
+                   ('clf', SVC(C=1.0,...))])
+    >>> rmtree(cachedir)
+
+.. warning:: **Side effect of caching on online/incremental transformers**
+
+   Enabling caching triggersx a clone of the transformers before fitting.
+   Therefore, the effect of past calls to ``fit`` is nullified and
+   the setting ``warm_start=True`` is ignored. The attributes of the
+   transformers will be equal to the cached transformers.
```

Comment:
I seem to remember that at some point, you had a small example of the right way of doing things. I think it should be added here, in the warning block. The reason is that this warning block is very abstract and general.